### PR TITLE
Add available_locales config option

### PR DESCRIPTION
**Pull request by akxcv**
This option allows to specify locales to use in Javascript. Defaults to `I18n.available_locales`. See #423

### CODE REVIEW

File: lib/i18n/js.rb [hunk 1]
```diff
@@ -188,6 +191,13 @@ def self.extract_segment_options(options)
       segment_options.merge(options.slice(*Segment::OPTIONS))
     end
 
+    def self.available_locales
+      config.fetch(:available_locales) do
+        # default value
+        I18n.available_locales rescue []
```
Comment (PikachuEXE) [inline lib/i18n/js.rb]: Why `rescue` is used here?

File: lib/i18n/js.rb [hunk 2]
```diff
@@ -149,10 +149,13 @@ def self.filter(translations, scopes)
...
     # Initialize and return translations
     def self.translations
-      ::I18n.backend.instance_eval do
+      backend_translations = ::I18n.backend.instance_eval do
         init_translations unless initialized?
-        translations.slice(*::I18n.available_locales)
+        translations
       end
+
+      return backend_translations if available_locales.empty?
```
Comment (PikachuEXE) [inline lib/i18n/js.rb]: Why do we want to return all backend translations when `available_locales` is empty?
We should remind them or at least return nothing (so developers notice they might have set something wrong)

**Comment by PikachuEXE** [comment]
I just merged #427 so some files contain conflicts now
Let me know if you are not sure how to fix
Thanks

File: lib/i18n/js/segment.rb [hunk 3]
```diff
@@ -3,24 +3,25 @@ module JS
...
-        @namespace    = options[:namespace] || 'I18n'
-        @pretty_print = !!options[:pretty_print]
-        @js_extend    = options.key?(:js_extend) ? !!options[:js_extend] : true
+        @file                  = file
+        @translations          = translations
+        @namespace             = options[:namespace] || 'I18n'
+        @pretty_print          = !!options[:pretty_print]
+        @js_extend             = options.key?(:js_extend) ? !!options[:js_extend] : true
         @sort_translation_keys = options.key?(:sort_translation_keys) ? !!options[:sort_translation_keys] : true
+        @available_locales     = options[:available_locales] || I18n.available_locales rescue []
```
Comment (akxcv) [inline lib/i18n/js/segment.rb]: Yeah, it's definitely duplication, but it's required in `segment`, too, see line 23.

File: lib/i18n/js.rb (hunk 1 above)
Comment (akxcv) [inline lib/i18n/js.rb]: If we get a `MethodMissing` (`I18n.available_locales`), we default to an empty array.

**Comment by akxcv** [comment]
Yeah, I'll fix the conflicts once we decide on the best way to add the `available_locales` option.

**Comment by akxcv** [comment]
Closing because no activity for several months

**Pull request closed without merging**